lumicks/pylake/detail/widefield.py

```diff
@@ -364,13 +383,39 @@ def __init__(self, tiff_file, align_requested):
                 self.json[f"Channel {j} alignment"] = self.json.pop(f"Alignment {color} channel")
                 self.json[f"Channel {j} detection wavelength (nm)"] = "N/A"
 
-        if "Channel 0 alignment" in self.json:
-            self._alignment_matrices = {
-                channel: TransformMatrix.from_alignment(
-                    self._raw_alignment_matrix(channel), *self.offsets
-                ).invert()
-                for channel in range(3)
-            }
+        excitation_colors = [
+            key.split(" ")[0] for key in self.json.keys() if "Excitation Laser wavelength" in key
+        ]
+
+        self.channel_order = (0, 1, 2)
+        if len(excitation_colors) == 2:
+            self.channel_order = tuple(
+                [
+                    ix
+                    for ix, color in enumerate(("Red", "Green", "Blue"))
+                    if color in excitation_colors
+                ]
+            )
+
+        excitation_filter_wavelengths = [
+            _parse_wavelength(self.json[f"Channel {channel} detection wavelength (nm)"])
+            for channel in range(3)
+            if f"Channel {channel} detection wavelength (nm)" in self.json
+        ]
+
+        if excitation_filter_wavelengths and all(excitation_filter_wavelengths):
+            # Sort by wavelength
+            self.channel_order = tuple(
+                self.channel_order[ix] for ix in np.argsort(excitation_filter_wavelengths)[::-1]
```

Comment:
```suggestion
                self.channel_order[ix] for ix in np.argsort(excitation_filter_wavelengths[::-1])
```
I would reverse the order of the `excitation_filter_wavelengths` list to indicate that you want the order of colors with decreasing wavelengths, i.e. "red", "green" and "blue".

Reply:
I don't think it's the same?

Let's say I have wavelengths `[2, 3, 1]`,  then `np.argsort(c)[::-1]` gives `[1, 0, 2]`, while `np.argsort(c[::-1])` gives `[0, 2, 1]`, but that's a moot point now since .

Reply:
Oh right, that was a quick shot :-(